src/Workspaces/Core/Portable/Workspace/Host/DocumentService/Extensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    internal static class Extensions
+    {
+        public static bool CanApplyChange(this TextDocument document)
+        {
+            return document?.State.CanApplyChange() ?? false;
+        }
+
+        public static bool CanApplyChange(this TextDocumentState document)
```

Comment:
Not really sure why we need this one and the above one.

Reply:
so I can use document.CanApplyChange? rather than document.Services.CanApplyChange?